fix: reinstate Liquid and remove Docker dependency

HashMapsData2Value opened the pull request:
Since we failed to upgrade liquid-auth-use-wallet-client from AlgoSDK v2.9.0 to v3 in time, Doug had the Liquid wallet temporarily removed from inclusion into Use-Wallet v4.

This is the revert PR: https://github.com/TxnLab/use-wallet/pull/337
This PR reverts the revert, while adjusting to changes to happened in the meantime (removal of networks). 

Due to issues I encountered with pnpm on my computer, I switched to using my own fork of liquid-auth-use-wallet. On it I merged Algosdk v3.1 and pushed version 1.1.0 to NPM, and then used that as an import.

Turns out, the only thing I really had to do was update the test. The src code itself was already AlgoSDK v3 proof. A small worry I had had (and admittedly the reason for not just taking a quick look at this earlier) was that there would be an issue with the Android client demo app itself, but I found through testing with this branch that it worked as normal. I leave it to the reviewer to double check that however.

To test: 

1. build the packages/use-wallet folder
2. build examples/vanilla-ts
3. cd into the liquid auth backend example and do docker-compose up -d
4. access ngrok page and test with Liquid Auth Android


Edit: We have decided to move away from using docker, and instead just use liquid-auth.on-render.com as the backend.

HashMapsData2Value committed:
Revert "refactor(liquid): remove Liquid Auth provider (#337)"

This reverts commit 74fdf73eb1e3c3dc764a26b419f6977a2201b413.

HashMapsData2Value committed:
fix: update liquid auth use wallet client version

chore: prettier

chore: fix test

chore: prettier

fix: replace with official @algorandfoundation

chore: remove unused ts expect error

chore: prettier fix

PhearZero reviewed (changes requested):
Let get this back to our repository for a full review, we can produce a canary when algorandfoundation/liquid-auth-use-wallet-client/pull/10 lands then just consume it directly. 

Without the artifact we will not be able to test properly, it should produce canary builds which we can use to test. Then the switch will be shallower to review

PhearZero reviewed (changes requested):
We will need to refactor the docker compose and default ports it looks like.

```yaml
services:
  liquid-auth:
    image: ghcr.io/algorandfoundation/liquid-auth:develop
    restart: no
    env_file:
      - .env.docker
    ports:
      - '3000:3000'
    depends_on:
      - redis
      - mongo

  ngrok:
    image: ngrok/ngrok:latest
    restart: no
    command:
      - 'start'
      - '--all'
      - '--config'
      - '/etc/ngrok.yml'
    volumes:
      - ./ngrok.yml:/etc/ngrok.yml
    ports:
      - 4040:4040
    depends_on:
      - liquid-auth

  redis:
    image: redis
    restart: always
    ports:
      - '6379:6379'

  mongo:
    image: mongo:7.0
    restart: no
    environment:
      - MONGO_INITDB_DATABASE=${DB_NAME:-fido}
      - MONGO_INITDB_ROOT_USERNAME=${DB_USERNAME:-algorand}
      - MONGO_INITDB_ROOT_PASSWORD=${DB_PASSWORD:-algorand}
    ports:
      - '27017:27017'
    volumes:
      - mongo:/data/db
volumes:
  mongo:

```

We also want to update the `.env.docker.example` and `.ngrok.yaml` examples to reflect a standard port like `3000`

HashMapsData2Value commented:
> We will need to refactor the docker compose and default ports it looks like.
> 
> ```yaml
> services:
>   liquid-auth:
>     image: ghcr.io/algorandfoundation/liquid-auth:develop
>     restart: no
>     env_file:
>       - .env.docker
>     ports:
>       - '3000:3000'
>     depends_on:
>       - redis
>       - mongo
> 
>   ngrok:
>     image: ngrok/ngrok:latest
>     restart: no
>     command:
>       - 'start'
>       - '--all'
>       - '--config'
>       - '/etc/ngrok.yml'
>     volumes:
>       - ./ngrok.yml:/etc/ngrok.yml
>     ports:
>       - 4040:4040
>     depends_on:
>       - liquid-auth
> 
>   redis:
>     image: redis
>     restart: always
>     ports:
>       - '6379:6379'
> 
>   mongo:
>     image: mongo:7.0
>     restart: no
>     environment:
>       - MONGO_INITDB_DATABASE=${DB_NAME:-fido}
>       - MONGO_INITDB_ROOT_USERNAME=${DB_USERNAME:-algorand}
>       - MONGO_INITDB_ROOT_PASSWORD=${DB_PASSWORD:-algorand}
>     ports:
>       - '27017:27017'
>     volumes:
>       - mongo:/data/db
> volumes:
>   mongo:
> ```
> 
> We also want to update the `.env.docker.example` and `.ngrok.yaml` examples to reflect a standard port like `3000`

Why would we need to make changes to the docker-compose? Did Liquid Auth change in the meanwhile?

PhearZero commented:
> Why would we need to make changes to the docker-compose? Did Liquid Auth change in the meanwhile?

Nothing has changed but we could improve the DX. We should be able to use the [development server in vite](https://liquidauth.com/server/integrations/#vite) as the endpoint. That would also remove the requirement of ngrok/https since everything will be `localhost`

HashMapsData2Value committed:
Merge branch 'v4' into fix/reinstate-liquid

HashMapsData2Value committed:
feat: replace custom liquid-auth backend with liquid-auth.onrender.com

HashMapsData2Value committed:
feat: replace custom liquid-auth backend with liquid-auth.onrender.com

docs: cover switch to liquid-auth.on-render.com

chore: remove unnecesary ignore path file

chore: prettier

HashMapsData2Value committed:
feat: replace custom liquid-auth backend with liquid-auth.onrender.com

docs: cover switch to liquid-auth.on-render.com

HashMapsData2Value commented on docs/fundamentals/get-started/README.md:
This file might have been deleted, and is now getting readied.

HashMapsData2Value commented on docs/fundamentals/supported-wallets.md:
Similar thing here, not sure if we are re-adding a file that was originally deleted but is now being re-added.

HashMapsData2Value commented:
Currently waiting for Michael to make changes to the liquid-auth backend (hosted at https://liquid-auth.onrender.com) to remove the strange bug we're getting now where it is complaining about "wrong wallet address".

HashMapsData2Value committed:
fix: bump liquid-auth-use-wallet-client

HashMapsData2Value committed:
fix: update rts username and credential configuration

HashMapsData2Value committed:
lint: prettier

HashMapsData2Value committed:
fix: resume session now calls disconnect

HashMapsData2Value committed:
chore: adjusted tests

lint: prettier

chore: typecheck

chore: prettier

HashMapsData2Value committed:
docs: include iOS 17 and update link

PhearZero reviewed (approved):
LGTM, lint/tests pass and the provider is working
